src/encrypt/mod.rs

```diff
@@ -358,6 +358,11 @@ impl crate::TaggedCborSerializable for CoseEncrypt0 {
 
 impl AsCborValue for CoseEncrypt0 {
     fn from_cbor_value(value: Value) -> Result<Self> {
+        const COSE_ENCRYPT0: u64 = iana::CborTag::CoseEncrypt0 as u64;
+        let value = match value {
+            Value::Tag(COSE_ENCRYPT0, v) => *v,
+            v => v,
+        };
```

Comment:
This doesn't seem like the right place for this, because it changes behaviour and isn't consistently applied across the various `struct`s of the crate.

Would the existing `TaggedCborSerializable` stuff cover what you want, either:
- as-is (e.g. by using `CoseEncrypt0::from_tagged_slice()`), or 
- by extending it (e.g. maybe add `{from,to}_tagged_cbor_value()` methods to the trait, with default implementations)?

Reply:
Frankly, I didn't see the `TaggedCborSerializable` parts; I was focused on the (de)serialization as used through `dcaf`, and might have been using it wrong.

I'll give it a try, and update this to reflect if there is any default method (eg. that strips a matching tag, if any) to add.

Thanks for the quick response!